lib/constants.js

```diff
@@ -1,25 +1,60 @@
 const path = require('path');
+const sanitizeFilename = require('sanitize-filename');
+const template = require('lodash-template');
+const { startsWith } = require('omnibelt');
 module.exports = {
   experience: {
     commandType: 'experience',
     apiType: 'experienceViews',
     localStatusParams: [ path.join(path.sep, '{components,layouts,pages}', '*.hbs') ],
     // eslint-disable-next-line no-template-curly-in-string
-    remoteStatusParams: [ path.join('experience', '${viewType}s', '${name}.hbs'), 'body' ]
+    remoteStatusParams: [
+      (resource, metaByFile) => {
+        const dir = path.join('experience', `${resource.viewType}s`);
+        const unsanitizeFilename = path.join(dir, `${resource.name}.hbs`);
+        if (metaByFile[unsanitizeFilename] && metaByFile[unsanitizeFilename].id === resource.id) {
+          // if the unsanitized version of the file already exists just keep it.
+          return unsanitizeFilename;
+        }
+        const filename = sanitizeFilename(resource.name);
+        if (!filename) {
+          // log a warning here..
+          return path.join(dir, `${resource.id}.hbs`);
+        }
+        const totalPath = path.join(dir, `${filename}.hbs`);
+        if (!metaByFile[totalPath] || metaByFile[totalPath].id === resource.id) {
+          return totalPath;
+        } else {
+          const allFiles = Object.keys(metaByFile);
+          const basicFilename = path.join(dir, `${filename}`);
+          let count = 0;
+          allFiles.forEach((localfileaname) => {
+            if (startsWith(basicFilename, localfileaname)) {
+              count++;
+            }
+          });
+          return `${basicFilename}-${count - 1}.hbs`;
```

Comment:
what if I literally manually named one of my views this name? For example:
I have the following "names":
`myView`
`myView-1`
`my:View`

it will attempt to name that last one `myView-1` which is already a used name.